rmc/utils/constraint.py

```diff
@@ -778,55 +775,54 @@ def get_rescue_1break_transcripts(
         the initial search pathway.
     :param rescue_threshold: Lower chi square significance threshold associated with
         the 'rescue' search pathway.
-    :return: None; function writes HT to resource path.
+    :return: Tuple set of sections (transcript_start_stop) with two simultaneous breaks and
+        set of sections with single breakpoint above 'rescue' threshold.
     """
     # Read in the simultaneous breaks results from initial search round 1
     simul_results_path = simul_search_round_bucket_path(
-        is_rescue=True,
+        is_rescue=False,
         search_num=1,
         bucket_type="final_results",
     )
     simul_ht = hl.read_table(
         f"{simul_results_path}/merged.ht",
     )
     simul_ht = simul_ht.annotate(transcript=simul_ht.section.split("_")[0])
-    simul_transcripts = simul_ht.aggregate(hl.agg.collect_as_set(simul_ht.transcript))
-
-    # Read in the single break temp chi square HT
-    chisq_ht = hl.read_table(f"{SINGLE_BREAK_TEMP_PATH}/all_loci_chisq.ht")
-    chisq_ht = chisq_ht.annotate(transcript=ht.section.split("_")[0])
-
-    # Filter to rows under initial threshold but over rescue threshold
-    # that were not found in the initial simultaneous breaks search
-    chisq_ht = chisq_ht.filter(
-        (chisq_ht.max_chisq < initial_threshold)
-        & (chisq_ht.max_chisq >= rescue_threshold)
-        & ~hl.literal(simul_transcripts).contains(chisq_ht.transcript)
-    )
-    chisq_ht = chisq_ht.checkpoint(
-        f"{TEMP_PATH_WITH_DEL}/rescue_chisq.ht", overwrite=True
-    )
+    simul_sections = simul_ht.aggregate(hl.agg.collect_as_set(simul_ht.sectopm))
 
-    # Get maximum chi square value per transcript
-    chisq_group_ht = chisq_ht.group_by("transcript").aggregate(
-        max_chisq=hl.agg.max(chisq_ht.max_chisq)
+    # Read in the no break found HT from single search round 1
+    # (of initial search)
+    ht = hl.read_table(
+        single_search_round_ht_path(
+            is_rescue=False,
+            search_num=1,
+            is_break_found=False,
+            is_breakpoint_only=False,
+        )
+    ).select_globals()
+
+    # Filter to transcripts that did not have simultaneous breakpoints
+    # in initial search
+    ht = ht.filter(
+        (ht.max_chisq < initial_threshold)
+        & ~hl.literal(simul_sections).contains(ht.section)
     )
-    chisq_group_ht = chisq_group_ht.checkpoint(
-        f"{TEMP_PATH_WITH_DEL}/rescue_single_break_max_chisq.ht", overwrite=True
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_DEL}/tmp_single_rescue_transcripts.ht", overwrite=True
     )
 
     # Annotate breakpoints of transcripts found with rescue threshold and checkpoint
-    breakpoint_ht = chisq_ht.annotate(
-        is_break=((chisq_ht.chisq == chisq_group_ht[chisq_ht.transcript].max_chisq))
+    breakpoint_ht = ht.annotate(
+        is_break=((ht.max_chisq >= rescue_threshold) & (ht.chisq == ht.max_chisq))
```

Comment:
Hm I think the filtering on rescue_threshold could be done in line ~816 unless you had a specific reason to wait until this line?

Reply:
nope that sounds good to me, this was just due to copy-pasting with moving code around